larray/core.py

```diff
@@ -3763,19 +3778,24 @@ class LArray(object):
     data : scalar, tuple, list or NumPy ndarray
         Input data.
     axes : collection (tuple, list or AxisCollection) of axes \
-    (int, str or  Axis), optional
+    (int, str or Axis), optional
         Axes.
     title : str, optional
         Title of array.
+    description : str, optional
+        Short description of the array.
+    metadata : OrderedDict(str, object), optional
+        Metadata associated with the array.
 
     Attributes
     ----------
     data : NumPy ndarray
         Data.
     axes : AxisCollection
         Axes.
-    title : str
-        Title.
+    metadata : OrdereDict(str, object)
```

Comment:
Alright. After all, if some users want to keep insertion order, they just need to upgrade to Python 3.6: https://mail.python.org/pipermail/python-dev/2016-September/146327.html
In that case, we can use a simple dictionary to store metadata.